blueprints/component-test/index.js

```diff
@@ -1,3 +1,65 @@
 module.exports = {
-  description: 'Generates a component unit test.'
+  description: 'Generates a component test.',
+
+  availableOptions: [
+    {
+      name: 'integration',
+      type: Boolean,
+      default: false,
+      aliases: ['i']
+    }
+  ],
+
+  locals: function(options) {
+    var prefix, testOpts, defaultTest;
+    if (options.integration) {
+      prefix = 'Integration: ';
+      testOpts = 'integration: true';
+      defaultTest = "// renders the component on the page\n" +
+             "       this.render('{{" + options.entity.name+ "}}');\n" +
+             "       expect(this.$()).to.be.ok;";
+    } else {
+      prefix = '';
+      testOpts = "// specify the other units that are required for this test\n" +
+             "    // needs: ['component:foo', 'helper:bar']";
+      defaultTest = "// creates the component instance\n" +
+              "      var component = this.subject();\n" +
+              "      expect(component._state).to.equal('preRender');\n\n" +
+              "      // renders the component on the page\n" +
+              "      this.render();\n" +
+              "      expect(component._state).to.equal('inDOM');";
+    }
+
+    return {
+      testPrefix: prefix,
+      testOpts: testOpts,
+      defaultTest: defaultTest,
+      integration: options.integration
+    };
+  },
+
+  fileMapTokens: function(options) {
+    return {
+      __testfolder__: function(options) {
+        if (options.locals.integration) {
+          return 'integration';
+        } else {
+          return 'unit';
+        }
+      }
+    };
+  },
+
+  afterInstall: function(options) {
+    if (!options.integration) { return; }
+
+    return this.insertIntoFile(
+      'Brocfile.js',
+      "// This is necessary because of component integration test.\n" +
+      "app.import({\n" +
+      "  test: app.bowerDirectory + '/ember/ember-template-compiler.js'\n" +
```

Comment:
I originally was going to use that, but I'm running into a problem with the auto-generated JSHint tests. The syntax supported by ember-cli-htmlbars-inline-precompile requires something like this:
```js
this.render(hbs`
  {{retailer-product-selector retailerProducts=retailerProducts}}
`);
```

which JSHint does not like - it produces this error:
```
should pass jshint
unit/components/foo-bar-test.js should pass jshint.
unit/components/foo-bar-test.js: line 49, col 26, Expected ')' and instead saw '{{foo-bar}}'.
unit/components/foo-bar-test.js: line 49, col 91, Missing semicolon.
unit/components/foo-bar-test.js: line 49, col 91, Expected an identifier and instead saw ')'.
```

Any suggestions on how to mitigate this? Can I make JSHint ignore this case?